src/components/layout/sync-button.tsx

```diff
@@ -0,0 +1,158 @@
+import { Button, Box, LinearProgress, Snackbar, Alert } from "@mui/material";
+import { DateTime } from "luxon";
+import { SyntheticEvent, useState } from "react";
+import { useApi } from "../../hooks/use-api";
+import { errorAtom } from "../../atoms/error";
+import { useAtom, useAtomValue, useSetAtom } from "jotai";
+import SyncDialog from "../contexts/sync-handler";
+import strings from "../../localization/strings";
+import { dailyEntriesAtom, personTotalTimeAtom, personsAtom } from "../../atoms/person";
+import { DailyEntry, Person } from "../../generated/client";
+import config from "../../app/config";
+import { userProfileAtom } from "../../atoms/auth";
+
+/**
+ * Sync button component
+ */
+const SyncButton = () => {
+  const { synchronizeApi } = useApi();
+  const setError = useSetAtom(errorAtom);
+  const yesterday: DateTime = DateTime.now().minus({ days: 1 });
+  const [syncStartDate, setSyncStartDate] = useState(yesterday);
+  const [syncing, setSyncing] = useState(false);
+  const [syncSuccess, setSyncSuccess] = useState(false);
+  const [syncHandlerOpen, setSyncHandlerOpen] = useState(false);
+  const [dailyEntries, setDailyEntries] = useAtom(dailyEntriesAtom);
+  const [persons, setPersons] = useAtom(personsAtom);
+  const { personsApi, dailyEntriesApi } = useApi();
+  const userProfile = useAtomValue(userProfileAtom);
+  const personTotalTime = useAtomValue(personTotalTimeAtom);
+
+  /**
+   * fetching persons and daily entries
+   */
+  const fetchPersonsAndEntries = async () => {
+    let tempPersons: Person[] = persons;
+    let fetchedDailyEntries: DailyEntry[] = [];
+  
+    if (!persons.length) {
+      try {
+        tempPersons = await personsApi.listPersons({});
+        setPersons(tempPersons);
+      } catch (error) {
+        setError(`${strings.error.fetchFailedGeneral}, ${error}`);
+      }
+    }
+  
+    if (tempPersons.length && personTotalTime) {
+      try {
+        const loggedInPerson = persons.find(
+          (person: Person) => person.keycloakId === userProfile?.id
+        );
+  
+        fetchedDailyEntries = await dailyEntriesApi.listDailyEntries({
+          personId: loggedInPerson?.id || config.person.forecastUserIdOverride
+        });
+        setDailyEntries(fetchedDailyEntries);
+      } catch (error) {
+        setError(`${strings.error.fetchFailedNoEntriesGeneral}, ${error}`);
+      }
+    }
+  
+    return fetchedDailyEntries;
+  };
+  
+  /**
+   * Get latest daily entry date
+   */
+  const getLatestDailyEntryDate = async () => {
```

Comment:
verb suggests that this is a get method but it does not actually return anything